stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/background/DefaultBackgroundDrawerImpl.kt

```diff
@@ -0,0 +1,108 @@
+        return ShapeAppearanceModel.builder()
+            .setAllCornerSizes(DEFAULT_CORNER_RADIUS)
+            .apply {
+                when {
+                    data.isBottomPosition() && data.isMine -> setBottomRightCornerSize(0f)
+                    data.isBottomPosition() && data.isTheirs -> setBottomLeftCornerSize(0f)
+                }
+            }
+            .build()
+            .let(::MaterialShapeDrawable)
+            .apply {
+                setTint(style.messageDeletedBackground)
+            }
+    }
+
+    override fun textAndAttachmentMessageBackground(context: Context, data: MessageListItem.MessageItem): Drawable {
+        return defaultBackground(context, data)
+    }
+
+    private fun defaultBackground(context: Context, data: MessageListItem.MessageItem): Drawable {
+        val radius = DEFAULT_CORNER_RADIUS
+        val bottomRightCorner = if (data.isMine && data.isBottomPosition()) 0f else radius
+        val bottomLeftCorner = if (data.isMine.not() && data.isBottomPosition()) 0f else radius
+
+        val shapeAppearanceModel = ShapeAppearanceModel.builder()
+            .setAllCornerSizes(radius)
+            .setBottomLeftCornerSize(bottomLeftCorner)
+            .setBottomRightCornerSize(bottomRightCorner)
+            .build()
+
+        return MaterialShapeDrawable(shapeAppearanceModel).apply {
+            val hasLink = data.message.attachments.any(Attachment::hasLink)
+            if (data.isMine) {
+                paintStyle = Paint.Style.FILL_AND_STROKE
+                setStrokeTint(style.messageStrokeColorMine)
+                strokeWidth = style.messageStrokeWidthMine
+                // for messages with links, we use a different background color than other messages by default.
+                // however, if a user has specified a background color attribute, we use it for _all_ message backgrounds.
+                val backgroundTintColor = if (hasLink) {
+                    style.messageLinkBackgroundColorMine
+                } else {
+                    style.messageBackgroundColorMine ?: ContextCompat.getColor(
+                        context,
+                        MESSAGE_CURRENT_USER_BACKGROUND
+                    )
+                }
+
+                setTint(backgroundTintColor)
+            } else {
+                paintStyle = Paint.Style.FILL_AND_STROKE
+                setStrokeTint(style.messageStrokeColorTheirs)
+                strokeWidth = style.messageStrokeWidthTheirs
+
+                val backgroundTintColor = if (hasLink) {
+                    style.messageLinkBackgroundColorTheirs
+                } else {
+                    style.messageBackgroundColorTheirs ?: ContextCompat.getColor(
+                        context,
+                        MESSAGE_OTHER_USER_BACKGROUND
+                    )
+                }
+
+                setTint(backgroundTintColor)
+            }
+        }
+    }
+
```

Comment:
It's a pity that returning Drawable here by wrapping the whole thing with `MaterialShapeDrawable` will be an overhead.

Reply:
Indeed, it doesn't draw anything. We could name it `MessageBackgroundFactory`, what do you think?